### ISSUE TITLE
linphone broken by upgrading soci; it builds but doesn't launch a GUI

### ISSUE THREAD

**Issue by mm--** [NixOS/nixpkgs#144577]
### Describe the bug
The current unstable version of linphone doesn't launch a GUI when run.
Instead, it runs and gets to a point where it says
```
[0xf4f800][Info]"Core created. Enable iterate."
```
and then does nothing.
It's not hanging, though, as `linphone show` can talk to the main `linphone` instance, but linphone still won't show a GUI.

### Screenshots

https://user-images.githubusercontent.com/3735202/140405497-b8dbf2cf-525d-4cc2-aae7-56c196de0e5e.mp4

### Steps To Reproduce
1. `nix run nixpkgs.linphone -c linphone`
2. See if a window ever appears

### Expected behavior
Linphone should launch a window when first run.
Or, `linphone show` should force a window to appear.

### Additional context
I tried git bisecting this to see what the problem is.
I know that commit f787c0b1304 is good.
The bisect seems to imply that one of the following commits causes linphone to stop working:
- 980f14081fdf8d58809be28ff5f94cce4fe25967 (#139887)
- 2287fbebc80f9c101f6406223e993471d3e1757c
- 5763d8c2f4dc60dbc85ddfc09a9bd22d220f2189

During the bisect I had to skip a few commits due to build failures.

Actually, I just built it with `soci 4.0.1` and `soci 4.0.2`. The upgrade of `soci` causes this bug.

### Notify maintainers
@jluttine

Also @Artturin who upgraded `soci` and `liblinphone`.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.14.14, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"home-manager, nixos-21.11pre130979.gfedcba.tar.xz, nixunstable-21.11pre297098.7e567a3d092"`
 - channels(me): `"home-manager-20.03, nixos-20.03-20.03.3328.6d1a044fc9f, nixos-20.09-20.09.4346.d6a5be86b6a, nixos-21.05-21.05.1153.0ccd0d91361, nixos-unstable-21.11pre297098.7e567a3d092"`
 - nixpkgs: `/home/me/github/nix/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- linphone
# a list of nixos modules affected by the problem
module:
```

### (Other questions I have)

If `soci 4.0.2` is incompatible with `linphone`, what's the preferred way of supplying it with the correct version?
Adding another attribute like `soci-for-linphone`?

Since `linphone` technically runs, but doesn't produce a GUI, it probably didn't fail any tests. Is there a way to test that a window shows up correctly?

**Comment by mm--** [comment]
Added a screen recorded to make things clearer.

In the meantime, I'm working around this using the following overlay:
```nix
self: super:
{
  soci = super.soci.overrideAttrs (old: rec {
    pname = "soci";
    version = "4.0.1";

    src = super.fetchFromGitHub {
      owner = "SOCI";
      repo = pname;
      rev = version;
      sha256 = "sha256-d4GtxDaB+yGfyCnbvnLRUYcrPSMkUF7Opu6+SZd8opM=";
    };
  });
}
```
This works well enough for me, but I don't have other packages (like Rstudio) that might require later versions of soci.

**Comment by Artturin** [comment]
i'll try updating linphone

**Comment by jbedo** [comment]
FWIW I can duplicate the issue with linphone, but Rstudio is fine.

**Comment by kenranunderscore** [comment]
Running into the same problem. I'll try the workaround above and see whether it helps me as well. I'm on unstable but my pin is a month old; is this maybe fixed now?

**Comment by c0bw3b** [comment]
Fixed on 21.11 with #157551
